tensorflow_io/azure/azfs_test.py

```diff
@@ -0,0 +1,262 @@
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+# ==============================================================================
+"""Tests for Azure File System."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+import tensorflow_io.azure.azfs_ops  # pylint: disable=unused-import
+from tensorflow.python.platform import test
+from tensorflow.python.platform import gfile
+
+class AZFSTest(test.TestCase):
+  """[summary]
+
+  Arguments:
+    test {[type]} -- [description]
+  """
+
+  def __init__(self, methodName='runTest'): # pylint: disable=invalid-name
+    self.account = os.environ.get(
+        'TF_AZURE_STORAGE_ACCOUNT', 'devstoreaccount1')
+    self.container = os.environ.get('TF_AZURE_STORAGE_CONTAINER', 'aztest')
+    self.path_root = 'az://' + os.path.join(self.account, self.container)
+    super(AZFSTest, self).__init__(methodName)
+
+  def _path_to(self, path):
+    return os.path.join(self.path_root, path)
+
+  def setUp(self): # pylint: disable=invalid-name
+    super(AZFSTest, self).setUp()
+    if not gfile.IsDirectory(self.path_root):
+      gfile.MakeDirs(self.path_root)
+
+  def tearDown(self): # pylint: disable=invalid-name
+    #   Cleanup container
+    if gfile.IsDirectory(self.path_root):
+      gfile.DeleteRecursively(self.path_root)
+
+  def test_exists(self):
+    self.assertTrue(gfile.IsDirectory(self.path_root))
+
+  def test_also_works_with_full_dns_name(self):
+    """Test the file system also works when we're given
+       a path of the form
+       az://<account>.blob.core.windows.net/<container>/<path>
+    """
+    file_name = self.account + '.blob.core.windows.net' + self.container
+    if not gfile.IsDirectory(file_name):
+      gfile.MakeDirs(file_name)
+
+    self.assertTrue(gfile.IsDirectory(file_name))
+
+  def test_create_file(self):
+    """Test create file.
+    """
+    # Setup and check preconditions.
+    file_name = self._path_to("testfile")
+    if gfile.Exists(file_name):
+      gfile.Remove(file_name)
+    # Create file.
+    with gfile.Open(file_name, 'w') as w:
+      w.write("")
+    # Check that file was created.
+    self.assertTrue(gfile.Exists(file_name))
+
+    gfile.Remove(file_name)
+
+  def test_write_read_file(self):
+    """Test write/read file.
+    """
+    # Setup and check preconditions.
+    file_name = self._path_to("writereadfile")
+    if gfile.Exists(file_name):
+      gfile.Remove(file_name)
+
+    rows = 100
+    # Write data.
+    with gfile.Open(file_name, 'w') as w:
+      for i in range(rows):
+        w.write("This is row {}\n".format(i))
+
+    # Read data.
+    with gfile.Open(file_name, 'r') as r:
+      for i, line in enumerate(r):
+        self.assertEqual("This is row {}\n".format(i), line)
```

Comment:
@yongtang any idea why this test might be failing? It writes the file as expected. When I download that file from Azure it's ascii format. When it does the read, it complains about not being able to decode 
`UnicodeDecodeError: 'utf-8' codec can't decode byte 0xbb in position 8: invalid start byte`

Reply:
@damienpontifex I can give it a try if there is a sample public azfs I could access (don't have azure account).

On a side note, maybe you could give `with gfile.Open(file_name, 'rb')` a try?